pyne/alara.py

```diff
@@ -205,16 +206,33 @@ def photon_source_hdf5_to_mesh(mesh, filename, tags):
         dictionary could be:
 
         tags = {('U-235', 'shutdown') : 'tag1', ('TOTAL', '1 h') : 'tag2'}
+    sub_voxel: bool, optional
+        If the sub_voxel is True, then the sub-voxel r2s will be used.
+        Then the photon_source will be interpreted as sub-voxel photon source.
+    cell_mats : dict, optional
+        cell_mats is required when sub_voxel is True.
+        Maps geometry cell numbers to PyNE Material objects.
     """
+
     # find number of energy groups
     with tb.open_file(filename) as h5f:
         num_e_groups = len(h5f.root.data[0][3])
+    max_num_cells = -1
+    ves = list(mesh.iter_ve())
+    if sub_voxel:
+        num_vol_elements = len(mesh)
+        subvoxel_array = _get_subvoxel_array(mesh, cell_mats)
+        max_num_cells = \
+                len(mesh.mesh.getTagHandle('cell_number')[ves[0]])
```

Comment:
If you only need the first element of ves, you don't need to convert the whole thing to a list and then index. You can just call `next()` on the iterator:  `ve0 = next(mesh.iter_ve())`

Reply:
@scopatz Thanks, I changed this part according to your suggestion.